mod_reforged/hooks/skills/perks/perk_mastery_cleaver.nut

```diff
@@ -0,0 +1,34 @@
+::Reforged.HooksMod.hook("scripts/skills/perks/perk_mastery_cleaver", function(q) {
+	q.onAdded = @(__original) function()
+	{
+		__original();
+		this.getContainer().add(::MSU.new("scripts/skills/perks/perk_rf_bloodlust", function(o) {
+			o.m.IsRefundable = false;
+			o.m.IsSerialized = false;
+		}));
+	}
+
+	q.onRemoved = @(__original) function()
+	{
+		__original();
+		this.getContainer().removeByID("perk.rf_bloodlust");
+	}
+
+	q.isEnabled <- function()
+	{
+		if (this.getContainer().getActor().isDisarmed())
+			return false;
+
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		return weapon != null && weapon.isWeaponType(::Const.Items.WeaponType.Cleaver);
+	}
+
+	q.onTargetHit = @(__original) function( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		__original(_skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor);
+		if (_targetEntity.isAlive() && _damageInflictedHitpoints >= ::Const.Combat.MinDamageToApplyBleeding && !_targetEntity.getCurrentProperties().isImmuneToBleeding && _skill.isAttack() && _skill.m.IsWeaponSkill && this.isEnabled())
```

Comment:
This doesn't check whether the skill is actually coming from the equipped cleaver. If someone has a cleaver equipped but attacks with another skill (probably only possible with mods) then those attacks will also inflict a bleed stack. E.g. Hand-To-Hand being always available. Or a kick ability that deals damage.

You could check `_skill.getItem()` of the passed skill.

But even then the cleaver could be a hybrid weapon with a piercing/blunt attack. Should that attack then also inflict bleeds?